### PR TITLE
Update privdrop, use OsString, add --chroot, build on !unix

### PR DESCRIPTION
**Pull request by Freaky**
**Choose one:** is this a 🐛 bug fix, a 🙋 feature, or a 🔦 documentation change?

... er, am I allowed to use all three?

* Update to privdrop 0.3.1 (fixes potential unsafety in threaded contexts).
* Switch to OsString for user and group name.
* Add `--chroot` support behind a feature flag.
* Adds a stub implementation with no flags for non-Unix platforms.
* Updates README and Bumps to 0.2.0.
* Fixes CI.

## Checklist

- [x] tests pass
- [x] tests and/or benchmarks are included (ish)
- [x] documentation is changed or added

## Context

jedisct1/rust-privdrop#9

## Semver Changes

This involves an API change (`user()` and `group()` now return `OsString`), hence the bump to 0.2.0.

### CODE REVIEW

**Comment by Dylan-DPC** [comment]
Thanks but i find it odd the way it is handled if we are not on unix. Instead of silently returning a default value, i'd prefer if we could just error out or something so that the user knows that it isn't supported on those platforms.

**Comment by Freaky** [comment]
Users should have a fair idea that it's not supported because the flags won't be defined.  Clap will exit with a usage error if they try to use them.

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "clap-permission-flag"
-version = "0.1.0"
+version = "0.2.0"
```
Comment (yoshuawuyts) [inline Cargo.toml]: Could you leave the version out of this patch? That way we can change the version as part of the release PR (:

File: README.md [hunk 2]
```diff
@@ -28,9 +28,12 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
 }
 ```
 
+This crate depends on Unix features - on other platforms a stub implementation
+which has no flags and always succeeds is used.
```
Comment (yoshuawuyts) [inline README.md]:
```suggestion
which has no flags and always succeeds if used.
```

File: Cargo.toml (hunk 1 above)
Comment (Freaky) [reply Cargo.toml]: Fixed in 8ccef1b

File: README.md (hunk 2 above)
Comment (Freaky) [reply README.md]: Fixed in 915fe16

**Comment by Freaky** [comment]
Thanks for the review, fixes pushed :)

**Comment by yoshuawuyts** [comment]
`v0.2.0` :tada:

**Pull request merged**